### PR TITLE
Adding matched track quality variables for pfcandAnalyzer

### PR DESCRIPTION
**Pull request by abaty**
Describe the Pull-Request:

This change adds an option to loop over the track collection and match tracks with their charged pf candidate partner in the pfcandAnalyzer tree.  The quality variables of the matched track are then output into the tree.  5 branches have been added.  The matching between pfcand and track should be exact (not done via eta/phi matching).

Does the PR pass the test script located at
HeavyIonsAnalysis/JetAnalysis/test/runtest.sh
[X] Yes
[] No

Please make sure to mention (using the @ syntax) anyone who might be interested in this PR.
@cfmcginn Could you please take a look at the distributions and make sure the matching looks reasonable to you?

### CODE REVIEW

File: HeavyIonsAnalysis/JetAnalysis/src/HiPFCandAnalyzer.cc [hunk 1]
```diff
@@ -85,6 +99,49 @@ HiPFCandAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetu
...
+        pfEvt_.trkChi2_.push_back( 0 );  
+        pfEvt_.trkNdof_.push_back( -999 );  
+        continue;
+      }
+
+      //find the track key
+      unsigned int trackKey = pfcand.trackRef().key();     
+
+      //loop through tracks looking for this key 
+      for(unsigned int it=0; it<tracks->size(); ++it){
```
Comment (bi-ran) [inline HeavyIonsAnalysis/JetAnalysis/src/HiPFCandAnalyzer.cc]: is the loop needed if the key is exactly the index?

**Comment by bi-ran** [comment]
thanks! @abaty 

there are no matched tracks for the cs pf candidates. can you please confirm that this is expected, and disable it for the pfcandAnalyzerCS tree if so?

**Comment by mverwe** [comment]
if we want track level info for CS PF candidates, you have to first go back to the original PF candidate using sourceCandidatePtr(0). However, the problem should be really solved prior to running the subtraction.

**Comment by abaty** [comment]
I've removed the track matching from the CS analyzer.  Also simplified the track key lookup to not require a loop.

One thing to note: charged hadrons and muons seem to have good track matching, but I am not sure if the track matching will work appropriately for electrons b/c they use a different track collection.  It would be good if an electron expert can investigate this, but I think that should not hold up this PR.

**Comment by bi-ran** [comment]
the gsf track collection is separate from the general tracks collection, but it is possible that the same seed is used for both, which will give the electron object a reference to the track in the general track collection. in general, there is no guarantee of a match existing.

@cfmcginn do you have any comments?

**Comment by abaty** [comment]
Hi,

Does anyone have any other objections to merging this?

@cfmcginn @bi-ran

**Comment by abaty** [comment]
Friendly reminder that this is still not merged.  @bi-ran

**Comment by bi-ran** [comment]
@abaty sorry for the delay

**Pull request merged**